src/Lucene.Net.ObjectMapping/Mapping/MappedFieldResolver.QueryVisitor.cs

```diff
@@ -378,41 +378,65 @@ private static TermQuery MakeTermQuery(MappedField field, Expression term)
                 Debug.Assert(null != field, "The mapped field must not be null.");
                 Debug.Assert(null != term, "The term must not be null.");
 
-                string queryTerm;
+                Term tm;
 
                 switch (field.Type)
                 {
                     case MappedField.FieldType.Float:
-                        queryTerm = NumericUtils.FloatToPrefixCoded(field.GetValueFromExpression<float>(term));
-                        break;
+                        {
+                            var bytes = new BytesRef(NumericUtils.BUF_SIZE_INT64);
+                            long l = NumericUtils.DoubleToSortableInt64(field.GetValueFromExpression<float>(term));
+                            NumericUtils.Int64ToPrefixCoded(l, 0, bytes);
+                            tm = new Term(field.Name, bytes);
+                            //queryTerm = NumericUtils.FloatToPrefixCoded(field.GetValueFromExpression<float>(term));
```

Comment:
Please do not check-in commented code. If it's used, it shouldn't be commented, if it isn't, it shouldn't be there.